library/src/main/kotlin/org/codice/compliance/SAMLComplianceException.kt

```diff
@@ -13,41 +13,52 @@
  */
 package org.codice.compliance
 
+import org.w3c.dom.Node
+
 class SAMLComplianceException : Exception {
 
     private constructor(message: String) : super(message)
     private constructor(message: String, cause: Throwable) : super(message, cause)
 
     companion object {
-        fun create(vararg codes: SAMLSpecRefMessage, message: String, cause: Throwable? = null):
+        fun create(vararg codes: SAMLSpecRefMessage,
+                   message: String,
+                   cause: Throwable? = null,
+                   node: Node? = null):
                 SAMLComplianceException {
             val samlExceptions = codes.map(Companion::readCode)
                     .fold("SAML Specification References:\n") { acc, s ->
                         "$acc\n$s"
                     }
 
-            if (cause != null) {
-                return SAMLComplianceException("$message\n\n$samlExceptions\n", cause)
+            return if (cause != null) {
+                SAMLComplianceException("$message\n\n$samlExceptions\n\n${node?.prettyPrintXml() ?: ""}\n", cause)
             } else {
-                return SAMLComplianceException("$message\n\n$samlExceptions\n")
+                SAMLComplianceException("$message\n\n$samlExceptions\n\n${node?.prettyPrintXml() ?: ""}\n")
             }
         }
 
-        fun createWithPropertyReqMessage(section: String, property: String, parent: String): SAMLComplianceException {
-            return SAMLComplianceException("$section: $property is required in $parent.")
-        }
-
-        fun createWithPropertyInvalidMessage(code: SAMLSpecRefMessage, property: String, propertyValue: String?):
-                SAMLComplianceException {
-            return SAMLComplianceException("The $property value of $propertyValue is invalid.\n\n${readCode(code)}")
+        fun createWithPropertyReqMessage(section: String,
```

Comment:
❓ Should we specify that `property` in this context means a property in an xml, and not some other property like in an object?

Maybe something like:
```
createWithXmlPropertyReqMessage
```

Reply:
I can make that change.